components/AccountWidget/AccountWidget.jsx

```diff
@@ -48,4 +49,20 @@ const mapDispatchToProps = (dispatch) => ({
   getPlayer: (playerId) => dispatch(getPlayer(playerId)),
 });
 
-export default connect(mapStateToProps, mapDispatchToProps)(AccountWidget);
+const getData = props => {
+  props.getPlayer(props.user.account_id);
+};
+
+class RequestLayer extends React.Component {
+  componentWillUpdate(nextProps) {
+    if (nextProps.user && nextProps.user.account_id) {
+      getData(nextProps);
+    }
```

Comment:
getPlayer is correct.  metadata only contains the user ID, but not the name/profile picture.